src/esperanza/validatorstate.h

```diff
@@ -17,56 +17,71 @@
 namespace esperanza {
 
 // clang-format off
+// All phases are ordered in a way how they can progress.
+// Every phase changes either one step up and down (if re-org happens).
+// Every phase has sparse index because if we want to introduce
+// a new one, we can include it in between without breaking
+// a layout on disk
```

Comment:
The whole comment describes the finite state machine whereas the proposed change eliminates visible fsm. I'd probably rephrase this somehow to not refer any phase progression.

Reply:
I removed a controvertial sentence `Every phase changes either one step up and down (if re-org happens).`. I kept the first sentence, you probably can question it too (feel free to re-open conversion) and kept the last one as it has nothing to do with finite state machine